include/boost/numeric/ublas/tensor/tensor.hpp

```diff
@@ -7,615 +7,193 @@
-    using value_type                = typename array_type::value_type;
-
-    using reference                 = typename array_type::reference;
-    using const_reference           = typename array_type::const_reference;
-
-    using pointer                   = typename array_type::pointer;
-    using const_pointer             = typename array_type::const_pointer;
-
-    using iterator                  = typename array_type::iterator;
-    using const_iterator            = typename array_type::const_iterator;
-
-    using reverse_iterator          = typename array_type::reverse_iterator;
-    using const_reverse_iterator    = typename array_type::const_reverse_iterator;
-
-    using tensor_temporary_type     = self_type;
-    using storage_category          = dense_tag;
-    using container_tag             = typename tensor_traits<T>::container_tag;
+namespace boost::numeric::ublas{
 
-    using extents_type              = typename tensor_traits<T>::extents_type;
-    using strides_type              = strides_t<extents_type,layout_type>;
+    namespace layout{
 
-    using matrix_type               = matrix<value_type,layout_type, std::vector<value_type> >;
-    using vector_type               = vector<value_type, std::vector<value_type> >;
+        template<typename...>
+        struct first_order;
+        
+        template<>
+        struct first_order<>{
+            
+            template<typename ExtentsType>
+            struct strides{
+                using extents_type = ExtentsType;
+                using layout_type = ::boost::numeric::ublas::first_order;
+                using strides_type = strides_t<ExtentsType,layout_type>;
+            };
+            
+        };
+        
+        template<typename ExtentsType>
+        struct first_order<ExtentsType>{
+            
+            template<typename>
+            struct strides{
+                using extents_type = ExtentsType;
+                using layout_type = ::boost::numeric::ublas::first_order;
+                using strides_type = strides_t<ExtentsType,layout_type>;
+            };
+
+        };
+
+        template<typename...>
+        struct last_order;
+
+        template<>
+        struct last_order<>{
+            
+            template<typename ExtentsType>
+            struct strides{
+                using extents_type = ExtentsType;
+                using layout_type = ::boost::numeric::ublas::last_order;
+                using strides_type = strides_t<ExtentsType,layout_type>;
+            };
+            
+        };
+        
+        template<typename ExtentsType>
+        struct last_order<ExtentsType>{
+            
+            template<typename>
+            struct strides{
+                using extents_type = ExtentsType;
+                using layout_type = ::boost::numeric::ublas::last_order;
+                using strides_type = strides_t<ExtentsType,layout_type>;
+            };
+
+        };
```

Comment:
I would like to have this in a separate `layout.hpp` file.

Reply:
ok.